pandas/tests/indexes/test_numpy_compat.py

```diff
@@ -66,7 +73,11 @@ def test_numpy_ufuncs_basic(index, func):
         if len(index) == 0:
             pass
         else:
-            with pytest.raises(Exception):
+            msg = (
+                r"loop of ufunc does not support argument 0 of type .* which "
+                f"has no callable {func.__name__} method"
+            )
+            with pytest.raises(TypeError, match=msg):
```

Comment:
Changes in this file look to have caused some failures (https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=50879&view=logs&j=b516280a-8c75-541b-8ba0-ae13561718a7&t=88a240e0-0f0f-5fd5-6f10-d82e025ed5d5)

Looks like `AttributeError` possible too. Since these errors are raised externally from `numpy`, might be more robust to also just use `external_error_raised` and not match on message in case of error message changes.

Reply:
Ok, it wasn't clear to me whether the error was raised in pandas or numpy. Is that also true of the other error in the same test? i.e. should I change both to `tm.external_error_raised`?

Reply:
Yep exactly, that's also an external error. I think both should use `tm.external_error_raised`.

Reply:
ok done. let's hope I fixed the tests.